SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -653,6 +656,237 @@ private void DisplayErrorMessage(string message)
+        {
+            LabelEmptyFieldsError.Visibility = Visibility.Collapsed;
+            LabelUsernameError.Visibility = Visibility.Collapsed;
+            LabelPasswordError.Visibility = Visibility.Collapsed;
+
+            var users = userLoginData.GetAllData();
+            var usernameExists = false;
+            string pswHash = null;
+            string pswSalt = null;
+            string username = null;
+            string userID = null;            
+
+            foreach (var name in users)
+            {
+                var checkUsernameExists = (name.Username == UsernameTextBox.Text);
+                if (checkUsernameExists)
+                {
+                    userID = name.ID;
+                    pswHash = name.PasswordHash;
+                    pswSalt = name.Salt;
+                    username = name.Username;
+                    usernameExists = true;
+                }
+            }
+
+            UserData registeredUser = new UserData(username, pswHash, pswSalt, userID);
+            UserData offlineUser = new UserData(UsernameTextBox.Text, null, pswSalt, userID);
+
+            if (UsernameTextBox.Text.Length == 0 || PasswordBox.Password.Length == 0)
+            {
+                LabelEmptyFieldsError.Visibility = Visibility.Visible;
+            }
+            else if (!usernameExists)
+            {
+                LabelUsernameError.Visibility = Visibility.Visible;
+            }
+            else if(registeredUser.PasswordHash != offlineUser.PasswordHash)
+            {
+                LabelPasswordError.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                ShowUserIDLabel.Content = "Your ID: " + userID;
+                ShowUsernameLabel.Content = username;
+
+                GridLogin.Visibility = Visibility.Collapsed; 
+                GridRegistration.Visibility = Visibility.Collapsed;
+            }
+
+            GridUserData.Visibility = Visibility.Collapsed;
+            GridUserDataToLogin.Visibility = Visibility.Collapsed;
+        }
+
+        private void RegisterButton_Clicked(object sender, RoutedEventArgs e)
+        {
+            var password = PasswordBoxR.Password;
+            var username = UsernameTextBoxR.Text;
+
+            LabelForUsedUsernameError.Visibility = Visibility.Collapsed;
+            LabelForInvalidUsernameError.Visibility = Visibility.Collapsed;
+            LabelForPasswordError.Visibility = Visibility.Collapsed;
+            LabelForConfirmPasswordError.Visibility = Visibility.Collapsed;
+            LabelForEmptyFieldsError.Visibility = Visibility.Collapsed;
+
+            if (CheckRegisterData(password, username))
+            {
+                userLoginData.AddData(new UserData(username));
+                userLoginData.SaveData();
+                
+                var users = userLoginData.GetAllData();
+                string userID = users[users.Count - 1].ID;
```

Comment:
Instead of trying to pull the ID out of the last user, you could store the "new UserData" in a variable, and then just get the ID from that.